### PR TITLE
feat: add base_url support for OpenAI provider in managed mode

### PR DESCRIPTION
**Pull request by avivhalfon**
> [!IMPORTANT]
> Add optional `base_url` support to `OpenAIProviderConfig` for custom API URL configuration in managed mode.
> 
>   - **Behavior**:
>     - Add `base_url` field to `OpenAIProviderConfig` in `dto.rs` for custom OpenAI API URL support.
>     - Update `transform_provider_dto()` in `config_provider_service.rs` to include `base_url` in parameters if present.
>   - **Misc**:
>     - No changes to existing functionality; only adds optional configuration support.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=traceloop%2Fhub&utm_source=github&utm_medium=referral)<sup> for 518fced6d4cc6258ff4b958d5e4cee72e582c89f. You can [customize](https://app.ellipsis.dev/traceloop/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

## Summary by CodeRabbit

* **New Features**
  * OpenAI and Azure providers now accept an optional custom base URL for alternative API endpoints; existing configs remain compatible.

* **Behavior**
  * When provided, base URLs are propagated into provider configuration parameters for use at request time.

* **Tests**
  * Test suites updated and a new integration test verifies creating and retrieving providers with a custom base URL.

### CODE REVIEW

File: src/management/services/config_provider_service.rs [hunk 1]
```diff
@@ -133,12 +133,23 @@ impl ConfigProviderService {
                 if let Some(org_id) = c.organization_id {
                     params.insert("organization_id".to_string(), org_id);
                 }
+                if let Some(base_url) = c
+                    .base_url
+                    .map(|s| s.trim().to_string())
```
Comment (nina-kollman) [inline src/management/services/config_provider_service.rs]: Is the trimming really necessary ? 
I cant see that we are doing it for any other param

**Review by nina-kollman — COMMENTED** [review]
Lets add test cases where the base_url is NOT none. The test should verify the correct value was chosen

File: src/management/services/config_provider_service.rs (hunk 1 above)
Comment (avivhalfon) [reply src/management/services/config_provider_service.rs]: i found it anoying when testing and code rabit suggested 
what do you think?

**Review by nina-kollman — APPROVED** [review]
